Realm/RLMSyncUtil_Private.h

```diff
@@ -22,6 +22,17 @@
 #import <Realm/RLMRealmConfiguration.h>
 #import <Realm/RLMSyncCredentials.h>
 
+typedef NS_ENUM(NSUInteger, RLMSyncSystemErrorKind) {
```

Comment:
I'm pretty sure that this enum is internal only. There should be explicit codes for the various kinds of errors (we don't map precisely from the object store categories, which is what this enum represents, to the user-facing ones for a couple of reasons).